src/v/cloud_storage/remote.cc

```diff
@@ -330,7 +338,7 @@ ss::future<upload_result> remote::upload_segment(
       segment_path,
       content_length);
     std::optional<upload_result> result;
-    while (!_gate.is_closed() && permit.is_allowed && !result) {
+    while (!_gate.is_closed() && permit.is_allowed && !result && is_leader()) {
```

Comment:
We probably should check after `_pool.acquire()` too.

Reply:
Added in latest version